### PR TITLE
Added user tracking

### PR DESCRIPTION
**Pull request by michalkowol**

### CODE REVIEW

File: src/api/e.php [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+$sessionId = pg_escape_string(@$json['sessionId']);
+$eventDatetime = pg_escape_string(@$json['eventDatetime']);
+$metadata = pg_escape_string(json_encode(@$json['metadata']));
+
+$connection = DatabaseHelper::createDatabaseConnection();
+pg_insert($connection, 'events', array('session_id' => $sessionId, 'event_datetime' => $eventDatetime, 'metadata' => $metadata));
+pg_close($connection);
+
+//HttpHelper::setNotContentResponseStatusCode();
```
Comment (michalkowol) [inline src/api/e.php]: Uncomment

File: src/api/mail.php [hunk 2]
```diff
@@ -0,0 +1,73 @@
+<?php declare(strict_types = 1);
+
+include_once('helpers/HttpHelper.php');
+include_once('helpers/JsonHelper.php');
+include_once('helpers/DatabaseHelper.php');
+
+class MailHelper {
```
Comment (michalkowol) [inline src/api/mail.php]: Move to new file

File: src/api/s.php [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+$sessionId = pg_escape_string(@$json['sessionId']);
+$clientId = $rawClientId ? pg_escape_string($rawClientId) : $rawClientId;
+$eventDatetime = pg_escape_string(@$json['eventDatetime']);
+$metadata = pg_escape_string(json_encode(@$json['metadata']));
+
+$connection = DatabaseHelper::createDatabaseConnection();
+pg_insert($connection, 'sessions', array('id' => $sessionId, 'client_id' => $clientId, 'event_datetime' => $eventDatetime, 'metadata' => $metadata));
+pg_close($connection);
+
+//HttpHelper::setNotContentResponseStatusCode();
```
Comment (michalkowol) [inline src/api/s.php]: Uncomment

File: src/api/t.php [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+$sessionId = pg_escape_string(@$json['sessionId']);
+$eventDatetime = pg_escape_string(@$json['eventDatetime']);
+$metadata = pg_escape_string(json_encode(@$json['metadata']));
+
+$connection = DatabaseHelper::createDatabaseConnection();
+pg_insert($connection, 'tracking', array('session_id' => $sessionId, 'event_datetime' => $eventDatetime, 'metadata' => $metadata));
+pg_close($connection);
+
+//HttpHelper::setNotContentResponseStatusCode();
```
Comment (michalkowol) [inline src/api/t.php]: Uncomment

File: src/db/changelog/changelog.xml [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+            <column name="session_id" type="uuid"/>
+            <column name="event_datetime" type="timestamp with time zone"/>
+            <column name="metadata" type="jsonb"/>
+        </createTable>
+        <addPrimaryKey tableName="events" columnNames="id"/>
+        <addDefaultValue tableName="events" columnName="id" defaultValueComputed="md5(random()::text || clock_timestamp()::text)::uuid"/>
+        <addNotNullConstraint tableName="events" columnName="session_id"/>
+        <addNotNullConstraint tableName="events" columnName="event_datetime"/>
+        <addNotNullConstraint tableName="events" columnName="metadata"/>
+    </changeSet>
```
Comment (michalkowol) [inline src/db/changelog/changelog.xml]: For the future - add FKs

Comment (pananimals) [reply src/db/changelog/changelog.xml]: Whats an FK?

Comment (michalkowol) [reply src/db/changelog/changelog.xml]: FK = Foreign key - https://en.wikipedia.org/wiki/Foreign_key - it is a concept from Relational Databases. It checks that relations are correct

**Pull request merged**